util/ranger/points.go

```diff
@@ -638,8 +638,6 @@ func (r *builder) buildFromNot(expr *expression.ScalarFunction) []*point {
 		startPoint := &point{value: types.MinNotNullDatum(), start: true}
 		endPoint := &point{value: types.MaxValueDatum()}
 		return []*point{startPoint, endPoint}
-	case ast.LogicAnd:
```

Comment:
Why do we remove this branch? we cannot guarantee that we have made all the necessary `PushDownNot` calls at least for now?

Reply:
https://github.com/pingcap/tidb/pull/22914#discussion_r602854476, this branch is not properly implemented. We need to implement `LogicOr`, `LT`, `GT` and so on. Shall we implement all these cases in this PR or just return `getFullRange()` for currently unimplemented cases?

Reply:
OK, we can leave the optimizations to a separate PR, and focus on the correctness problem in this PR.